schema.go

```diff
@@ -0,0 +1,110 @@
+package dalec
+
+import (
+	"bytes"
+	"encoding/json"
+	"slices"
+	"strings"
+
+	"github.com/atombender/go-jsonschema/pkg/schemas"
+	"github.com/invopop/jsonschema"
+)
+
+// GenerateJSONSchema generates and returns the JSON schema for Dalec specs.
+// This schema can be used by editors and tools for validation and autocomplete.
+func GenerateJSONSchema() ([]byte, error) {
```

Comment:
I don't really see a need to export this, can we move this back into the cmd main package?

Reply:
updated